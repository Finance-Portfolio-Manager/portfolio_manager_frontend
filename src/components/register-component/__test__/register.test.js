import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Register from './../register';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = mount(<Register></Register>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<Register></Register>);
    expect(tree).toMatchSnapshot();
})

describe("testing each function in register for email input", ()=>{
    afterEach(cleanup);

    test("handleChange function", ()=>{
        const emailInput = wrapper.find("FormControl").first();
        const changeEvent = ()=>{
            emailInput.simulate('change', {target: {value: 'trgreenman88@gmail.com'}});
        }
        expect(changeEvent).not.toThrow();
    })

    test("click Register submit button", ()=>{
        const submitButton = wrapper.find("button");
        expect(submitButton.text()).toContain("Register");
        const clickEvent = ()=>{
            submitButton.simulate('click');
        }
        expect(clickEvent).not.toThrow();
    })

    test("enter all valid fields then click submit", ()=>{
        const submitButton = wrapper.find("button");
        const emailInput = wrapper.find("FormControl").first();
        const usernameInput = wrapper.find("FormControl").at(1);
        const passwordInput = wrapper.find("FormControl").at(2);
        const passConfInput = wrapper.find("FormControl").last();
        const submitAllFields = ()=>{
            emailInput.simulate('change', {target: {value: 'trgreenman88@gmail.com', name: 'email'}});
            usernameInput.simulate('change', {target: {value: 'trgreenman88', name: 'username'}});
            passwordInput.simulate('change', {target: {value: 'Password123!', name: 'password'}});
            passConfInput.simulate('change', {target: {value: 'Password123!', name: 'confirmPassword'}});
            submitButton.simulate('click');
        }
        expect(submitAllFields).not.toThrow();
    })

    test("enter invalid email then click submit", ()=>{
        const submitButton = wrapper.find("button");
        const emailInput = wrapper.find("FormControl").first();
        const usernameInput = wrapper.find("FormControl").at(1);
        const passwordInput = wrapper.find("FormControl").at(2);
        const passConfInput = wrapper.find("FormControl").last();
        const submitAllFields = ()=>{
            emailInput.simulate('change', {target: {value: 'asdfadsf', name: 'email'}});
            usernameInput.simulate('change', {target: {value: 'trgreenman88', name: 'username'}});
            passwordInput.simulate('change', {target: {value: 'Password123!', name: 'password'}});
            passConfInput.simulate('change', {target: {value: 'Password123!', name: 'confirmPassword'}});
            submitButton.simulate('click');
        }
        expect(submitAllFields).not.toThrow();
    })

    test("enter short password (<8) then click submit", ()=>{
        const submitButton = wrapper.find("button");
        const emailInput = wrapper.find("FormControl").first();
        const usernameInput = wrapper.find("FormControl").at(1);
        const passwordInput = wrapper.find("FormControl").at(2);
        const passConfInput = wrapper.find("FormControl").last();
        const submitAllFields = ()=>{
            emailInput.simulate('change', {target: {value: 'trgreenman88@gmail.com', name: 'email'}});
            usernameInput.simulate('change', {target: {value: 'trgreenman88', name: 'username'}});
            passwordInput.simulate('change', {target: {value: 'short', name: 'password'}});
            passConfInput.simulate('change', {target: {value: 'short', name: 'confirmPassword'}});
            submitButton.simulate('click');
        }
        expect(submitAllFields).not.toThrow();
    })

    test("enter password with no special character then click submit", ()=>{
        const submitButton = wrapper.find("button");
        const emailInput = wrapper.find("FormControl").first();
        const usernameInput = wrapper.find("FormControl").at(1);
        const passwordInput = wrapper.find("FormControl").at(2);
        const passConfInput = wrapper.find("FormControl").last();
        const submitAllFields = ()=>{
            emailInput.simulate('change', {target: {value: 'trgreenman88@gmail.com', name: 'email'}});
            usernameInput.simulate('change', {target: {value: 'trgreenman88', name: 'username'}});
            passwordInput.simulate('change', {target: {value: 'Password123', name: 'password'}});
            passConfInput.simulate('change', {target: {value: 'Password123', name: 'confirmPassword'}});
            submitButton.simulate('click');
        }
        expect(submitAllFields).not.toThrow();
    })

    test("enter mismatched passwords then click submit", ()=>{
        const submitButton = wrapper.find("button");
        const emailInput = wrapper.find("FormControl").first();
        const usernameInput = wrapper.find("FormControl").at(1);
        const passwordInput = wrapper.find("FormControl").at(2);
        const passConfInput = wrapper.find("FormControl").last();
        const submitAllFields = ()=>{
            emailInput.simulate('change', {target: {value: 'trgreenman88@gmail.com', name: 'email'}});
            usernameInput.simulate('change', {target: {value: 'trgreenman88', name: 'username'}});
            passwordInput.simulate('change', {target: {value: 'Password123!', name: 'password'}});
            passConfInput.simulate('change', {target: {value: 'Password123!!!!', name: 'confirmPassword'}});
            submitButton.simulate('click');
        }
        expect(submitAllFields).not.toThrow();
    })

})