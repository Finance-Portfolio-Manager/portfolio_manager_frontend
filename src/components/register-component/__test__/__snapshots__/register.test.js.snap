// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <form
    className=""
  >
    <div
      className="aroundLogo"
    >
      <img
        alt=""
        className="logo"
        src="revlogo.jpg"
      />
    </div>
    <div
      className="form-inner"
    >
      <h4>
        Portfolio Manager
      </h4>
      <h6>
        <em>
          Account Registration
        </em>
      </h6>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="form-email"
        >
          Email:
        </label>
        <input
          className="form-control"
          id="form-email"
          name="email"
          onChange={[Function]}
          placeholder="Enter your email"
          required={true}
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="form-username"
        >
          Username:
        </label>
        <input
          className="form-control"
          id="form-username"
          name="username"
          onChange={[Function]}
          placeholder="Enter your username"
          required={true}
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="form-password"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="form-password"
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="form-confirm-password"
        >
          Confirm Password:
        </label>
        <input
          className="form-control"
          id="form-confirm-password"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Re-enter your password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="submitButton mt-2"
        onClick={[Function]}
        type="submit"
      >
        Register
      </button>
    </div>
  </form>
</div>
`;
